PictureApp.API/Controllers/UsersController .cs

```diff
@@ -15,25 +15,73 @@ public class UsersController : ControllerBase
     {
         private readonly IUserService _userService;
         private readonly IMapper _mapper;
+        private readonly IPhotoService _photoService;
 
-        public UsersController(IUserService userService, IMapper mapper)
+        public UsersController(IUserService userService, IMapper mapper, IPhotoService photoService)
         {
             _userService = userService ?? throw new ArgumentNullException(nameof(userService));
             _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            _photoService = photoService;
         }
 
         [HttpGet("{id}")]
-        public IActionResult GetUser(int id)
+        public async Task<IActionResult> GetUser(int id)
         {
-            return Ok(_userService.GetUser(id));
+            try
+            {
+                CheckIfCurrentUserIsCorrect(id);
+
+                return Ok(await _userService.GetUser(id));
+            }
+            catch(Exception)
+            {
+                return BadRequest();
+            }
         }
 
-        [HttpGet("allUserWithFollowerInfo")]
-        public async Task<IActionResult> GetUsersWithFollowers()
+        [HttpGet("userEditProfile/{id}")]
+        public async Task<IActionResult> GetUserForEdit(int id)
+        {
+            try 
+            {
+                CheckIfCurrentUserIsCorrect(id);
+
+                return Ok(await _userService.GetUserForEdit(id));
+            }
+            catch(Exception)
+            {
+                return BadRequest();
+            }
+
+        }
+
+        [HttpGet("getCurrentUserFollowersForDashboard/{id}")]
+        public async Task<IActionResult> GetUsersWithFollowersForCurrentUser(int id)
+        {
+            try
+            {
+                CheckIfCurrentUserIsCorrect(id);
```

Comment:
If we get any data for current user why do want to check whether current user is correct? I mean, here You can simply use User static class to obtain current user. If I understand correctly `id` parameter is not necessary in this method.